staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -742,16 +747,35 @@ func validateCustomResourceDefinitionValidation(ctx context.Context, customResou
 	return allErrs
 }
 
+type costInfo struct {
+	// Cardinality tracks the number of times a rule in the current schema node will actually get executed
+	// due to being the child of arrays/maps/etc.
+	Cardinality *uint64
```

Comment:
`MaxCardinality`?

Reply:
Yes, `Cardinality` isn't quite descriptive enough compared to what it's actually being used for - fixed, thanks!